orttraining/orttraining/test/graph/gradient_graph_builder_test.cc

```diff
@@ -798,6 +800,156 @@ class PipelineSplitter {
   }
 };
 
+// TODO: move to a proper location for pipeline training
+struct PipelineBatchInfo {
+  std::vector<std::pair<int64_t, int64_t>> events;
+  int64_t retired_batch;  // index of retired batch, so its data could be reused
+};
+
+class PipelineBatchPlanner {
+ private:
+  int64_t max_id_;
+
+ public:
+  PipelineBatchPlanner()
+      : max_id_(::onnxruntime::contrib::OrtEventPool::GetPoolSize() - 1) {
+  }
+
+  void CreatePlan(const std::vector<int64_t>& start_ids,
```

Comment:
For non-trivial public functions, we need to document their arguments and have example.